ui/candidate-portal/src/app/components/common/candidate-language-card/candidate-language-card.component.ts

```diff
@@ -35,21 +35,29 @@ export class CandidateLanguageCardComponent {
   @Input() languageLevels: LanguageLevel[];
 
   @Output() onDelete = new EventEmitter();
+  translatedLanguageName: string = '';
 
   constructor() { }
-
   delete() {
     this.onDelete.emit();
   }
+  // Watch for changes in the inputs and update the language name accordingly
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.language || changes.languages) {
+      this.getLanguageName();
+    }
+  }
 
-  getLanguageName(language) {
+  // Compute the language name once, when inputs change
+  getLanguageName() {
```

Comment:
I suggest a bit of refactoring - including changing the name of this method to lookupLanguageName.

"get" and "set" have very specific semantics associated with them. "get" implies that you are returning a value - which the previous implementation of the method did. However it no longer returns a value.  Instead it looks up a name given a language.

I would change the method to 

lookupLanguageName(language: CandidateLanguage): string

Then in the ngOnChanges code I would add:

this.translatedLanguageName = lookupLanguageName(this.language)

Then the implementation code inside the method just accesses the this.languages array rather than directly accessing the this.language and this.translatedLanguageName fields - which makes it more modular and more loosely coupled (and simpler to unit test)

Functionally no difference of course.

Reply:
@camerojo Thanks for the feedback.
Please have a look on new changes and let me know if there is something else.